File: simcoe/src/main/java/com/prolificinteractive/simcoe/WriteEvent.java
package com.prolificinteractive.simcoe;

import com.prolificinteractive.simcoe.trackers.AnalyticsTracking;

/**
 * The write event.
 */
class WriteEvent {

  /**
   * The analytics tracking provider.
   */
  final AnalyticsTracking provider;

  /**
   * The tracking result.
   */
  final TrackingResult trackingResult;

  /**
   * Creates a write event instance.
   *
   * @param provider The analytics tracking provider.
   * @param trackingResult The tracking result.
   */
  protected WriteEvent(final AnalyticsTracking provider, final TrackingResult trackingResult) {
    this.provider = provider;
    this.trackingResult = trackingResult;
  }
}

File: simcoe/src/main/java/com/prolificinteractive/simcoe/WriteEvent.kt
package com.prolificinteractive.simcoe

import com.prolificinteractive.simcoe.trackers.AnalyticsTracking

/**
 * The write event.
 *
 * @param provider The analytics tracking provider.
 * @param trackingResult The tracking result.
 */
internal class WriteEvent(
    val provider: AnalyticsTracking,
    val trackingResult: TrackingResult
)
